### DISCUSSION TITLE
NER custom model giving empty result

### DISCUSSION THREAD

**Discussion by kirankumari999** [explosion/spaCy#7803] [Help: Coding & Implementations]
model = None # Since we training a fresh model not a saved model
n_iter = 20
with nlp.disable_pipes(*other_pipes):  # only train ner
    # optimizer = nlp.begin_training()
    if model is None:
        optimizer = nlp.begin_training()
    else:
        optimizer = nlp.resume_training()
    for iteration in range(10):
        random.shuffle(TRAINING_DATA)
        losses = {}
        for text, entities in TRAINING_DATA:
            doc = nlp.make_doc(text)
            entity_offsets = entities.get("entities")
            gold = GoldParse(doc, entities=entity_offsets)
            nlp.entity.update([doc], [gold], drop=0.5, sgd=optimizer, losses=losses)
            print('Losses with gold', losses)
osses with gold {'ner': 304421.0290137643}
Losses with gold {'ner': 304642.6718746537}
Losses with gold {'ner': 304762.8316188568}
Losses with gold {'ner': 305164.48968632164}
Losses with gold {'ner': 305597.89806015434}
Losses with gold {'ner': 306354.5482810253}
Losses with gold {'ner': 306387.7921955618}
Losses with gold {'ner': 306948.703250173}
Losses with gold {'ner': 307005.35394229135}
Losses with gold {'ner': 307130.0234811469}
Losses with gold {'ner': 307613.8945380612}
Losses with gold {'ner': 307997.0952721997}
Losses with gold {'ner': 308148.7388317748}
Losses with gold {'ner': 308614.81227232656}
Losses with gold {'ner': 308947.34911085805}
Losses with gold {'ner': 309503.37237383565}
Losses with gold {'ner': 309728.1373324557}
Losses with gold {'ner': 309763.1743289656}
Losses with gold {'ner': 309950.049087686}
Losses with gold {'ner': 310222.3440226144}
Losses with gold {'ner': 310411.97120605176}
Losses with gold {'ner': 312247.0951760835}

doc = nlp_updated("Fridge can be ordered in CEO" )
print("Entities", [(ent.text, ent.label_) for ent in doc.ents])
Output: 
Entities []

As above we can see everything is working perfectly but still model is not able to predict. I am not able to figure out the issue in it. If i get any error so i can resolve it. Can anyone help me please

**Comment by svlandeg** [comment]
Hi! As a maintenance note: this is a good question for the discussions forum, so I'll move it there. The original issue will be closed and will link through to the open discussion thread.

**Comment by svlandeg** [comment]
Can you provide a bit more background information on your training data? How big is it, how many labels does it contain, perhaps some sample data for us to eyeball?

In general what I'd recommend is this: take just a handful examples and create a very small training dataset. Train on these examples to ensure your model can overfit on them. Now run your evaluation on the SAME small set of examples - this should give you (artificially) high performance scores near 100%. If that works, at least you know that your model & evaluation is set up correctly.

Then as a next step, it's probably worth digging a bit more into the data (cf my first questions) to understand what may be going on.

**Comment by svlandeg** [comment]
PS: as another maintenance note, could you please markup your posts to make them more readable for us? cf https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting, or use the menu on the top when editing/drafting a post/reply. Thanks!
